### DISCUSSION TITLE
Consider changing a license?

### DISCUSSION THREAD

**Discussion by TempleLin** [enable3d/enable3d#247] [General]
I just discovered this library and I must say it's the most simplistic yet powerful 3D physics game library I've seen. But the GPL license is not ideal for game development. Ever consider changing the license to something more suitable?

**Comment by yandeu** [comment]
Thanks!

What's wrong with the GPL?

**Reply by TempleLin** [reply]
Code built using GPL Licensed library must also be distributed under GPL license and therefore will be needed to be open source as well and everyone can  be able to use and extend it. Proprietary games cannot be allowed under this license. MIT or BSD license are more suitable for game development libraries. Proprietary games will be allowed to distribute with libraries of these licenses as long as they include the license notice.

**Reply by yandeu** [reply]
I don't think that is true. If you include enable3d as a library (for example via npm), you can use it however you want.

Only if you fork the project and make changes, you have to publish these changes.

You don't really distribute the code if you publish a game. The code is not visible to the gamer, and the source is minified and hard to read.

**Reply by TempleLin** [reply]
Okay seems like I'm mistaken. Please excuse my ignorance.

**Reply by yandeu** [reply]
No. Maybe I am wrong. Who knows. I'm not a lawyer. But I believe I am right.

**Reply by TempleLin** [reply]
Regardless, thank you for making this amazing library. It's so underrated!

**Reply by Dev-Punk** [reply]
> If you include enable3d as a library (for example via npm), you can use it however you want.
> 
> Only if you fork the project and make changes, you have to publish these changes.

That's LGPL. 
See: https://www.wikiwand.com/en/GNU_Lesser_General_Public_License

**If that's your intention please consider switching to LGPL.**

With GPL that's a gray area for Javascript libraries and because of that... many people tend to see that as it doesn't matter if a client side app (or a game) is closed source. If the app is using a GPL'd library then it has to be released as an open source GPL'd app.

See: 
https://duckduckgo.com/?q=javascript+gpl+lgpl
https://duckduckgo.com/?q=differences+between+lgpl+and+gpl

**Reply by yandeu** [reply]
Yes, I guess LGPL is what I want.

**Reply by yandeu** [reply]
Done.
